cesium/channel.go

```diff
@@ -111,3 +111,67 @@ func (db *DB) validateNewChannel(ch Channel) error {
 	}
 	return v.Error()
 }
+
+// RenameChannel changes the key of channel oldKey into newKey. This operation is
+// idempotent and does not return an error if the channel does not exist.
+// RenameChannel returns an error if there are open iterators/writers on the given channel.
+func (db *DB) RenameChannel(oldKey ChannelKey, newKey core.ChannelKey) error {
+	db.mu.Lock()
+	defer db.mu.Unlock()
+	udb, uok := db.unaryDBs[oldKey]
+	if uok {
+		if err := udb.TryRekey(); err != nil {
+			return err
+		}
+
+		err := db.fs.Rename(keyToDirName(oldKey), keyToDirName(newKey))
+		if err != nil {
+			return err
+		}
+
+		udb.Config.Channel.Key = newKey
```

Comment:
Not a tremendous fan of editing the internal properties of a class, but this seems ok.